src/orquestra/quantum/api/circuit_runner_contracts.py

```diff
@@ -0,0 +1,204 @@
+################################################################################
+# © Copyright 2022 Zapata Computing Inc.
+################################################################################
+import numpy as np
+
+from .circuit_runner import CircuitRunner
+from ..circuits import RX, RY, RZ, Circuit, H, CNOT
+
+
+_example_circuits = (
+    Circuit([RY(np.pi / 5)(0)]),
+    Circuit([H(0), CNOT(0, 1)]),
+    Circuit([H(0), CNOT(0, 2), H(1)]),
+    Circuit([RX(np.pi / 2)(0), H(2), RZ(np.pi / 3)(1), CNOT(0, 3)]),
+)
+
+_example_n_samples = (5, 10, 15, 20)
+
+_invalid_n_samples = (0, -1, -10)
+
+
+class _ValidateRunAndMeasure:
+    """Contracts relating to run_and_measure."""
+
+    @staticmethod
+    def returns_number_of_measurements_greater_or_equal_to_n_samples(
```

Comment:
Consider the following scenario.
- You have one circuit
- Qiskit backend used has a limit of N samples per experiment
- You want N+1 samples
- You do `runner.run_and_measure(my_circuit, N+1)`
- To accommodate, inner qiskit backend runs two experiments in a single run, with `shots=N`
- Number of shots is the same for all experiments, hence you get 2 * N samples >= N+1

Reply:
🤦
Shouldn't we just have an "equal" case in the contracts and then overwrite this particular contract for qiskit and add an explanation?
It sounds like an edge case to me?

Reply:
I would still put this extra case as another set of contracts as, dunno, "optional contracts"?
Otherwise  someone might forget about including it separately.

Reply:
[e0bf40d](https://github.com/zapatacomputing/orquestra-quantum/pull/66/commits/e0bf40dd265b76a9ee3a637c434d9ab8b04e1abc) I went with "STRICT_CIRCUIT_RUNNER_CONTRACTS", wdyt?

Reply:
Looks good!